app/apollo/resolvers/channel.js

```diff
@@ -414,29 +411,57 @@ const channelResolvers = {
           throw new RazeeValidationError(context.req.t('{{serSubCount}} service subscriptions depend on this channel version. Please have them updated/removed before removing this channel version.', {'serSubCount':serSubCount}), context);
         }
 
-        const channel_uuid = deployableVersionObj.channel_id;
-        const channel = await models.Channel.findOne({ uuid: channel_uuid, org_id });
+        // Get the Version
+        const deployableVersionObj = await models.DeployableVersion.findOne({ org_id, uuid });
+
+        // Get the Channel for the Version
+        let channel;
+        if( deployableVersionObj ) {
+          channel = await models.Channel.findOne({ uuid: deployableVersionObj.channel_id, org_id });
+        }
+        else {
+          channel = await models.Channel.findOne({ versions: { $elemMatch: { uuid: uuid } }, org_id });
+        }
         if(!channel){
-          throw new NotFoundError(context.req.t('channel uuid "{{channel_uuid}}" not found', {'channel_uuid':channel_uuid}), context);
+          // If unable to find the Channel then cannot verify authorization, so throw an error
+          throw new NotFoundError(context.req.t('channel for version "{{uuid}}" not found', {'uuid':uuid}), context);
```

Comment:
The whole procedure seems a little too complex and prints incorrect message about channel not being found when the real problem is Version does not exist. Please check if the following fix makes sense: 1. Change message from "channel for version "{{uuid}}" not found" -> "version uuid "{{uuid}}" not found" 2. Remove the last "if( !foundVersion ) {" as the exception in that block will probably never be thrown. 3. remove this foundVersion flag.

Reply:
1. Clarified the NotFoundError message to: `version uuid "{{uuid}}" not found and no references found`
2. Eliminated use of `foundVersion` -- you are correct, it will never enter the 'if' if it can get past the NotFoundError